pkg/nsx/services/firewall.go

```diff
@@ -726,3 +727,76 @@ func buildPeerTags(obj *v1alpha1.SecurityPolicy, peers *[]v1alpha1.SecurityPolic
 	}
 	return peerTags
 }
+
+func (service *SecurityPolicyService) deleteGroup(groupsClient domains.GroupsClient, nsxGroup *model.Group) error {
+	failIfSubtreeExistsParam := false
+	forceParam := false
+	return groupsClient.Delete(getDomain(), *nsxGroup.Id, &failIfSubtreeExistsParam, &forceParam)
+}
+
+func (service *SecurityPolicyService) DeleteSecurityPolicy(UID types.UID) error {
+	groupsClient := service.NSXClient.GroupClient
+	policiesClient := service.NSXClient.SecurityClient
+
+	policy, err := service.SecurityPolicyStore.ByIndex(util.TagScopeSecurityPolicyCRUID, string(UID))
+	if err != nil {
+		log.Error(err, "failed to find NSX SecurityPolicy from store", "UID", string(UID))
+		return err
+	}
+	if len(policy) == 0 {
+		log.V(1).Info("no NSX SecurityPolicy is found", "UID", string(UID))
+		return nil
+	}
+	nsxSecurityPolicy := policy[0].(model.SecurityPolicy)
+	err = policiesClient.Delete(getDomain(), *nsxSecurityPolicy.Id)
+	if err != nil {
+		log.Error(err, "failed to delete NSX SecurityPolicy", "UID", string(UID))
+		return err
+	}
+	service.SecurityPolicyStore.Delete(nsxSecurityPolicy)
+
+	rules, err := service.SecurityPolicyStore.ByIndex(util.TagScopeSecurityPolicyCRUID, string(UID))
+	if err != nil {
+		log.Error(err, "failed to find NSX Rules from store", "UID", string(UID))
```

Comment:
ditto. In https://github.com/kubernetes/client-go/blob/8c8fa70f7a2acc191d4787327639621f69992efd/tools/cache/thread_safe_store.go#L89 , the store will check if the element exists before invoking `delete`, so I think it's worthy to try to directly call `Delete`

Reply:
search is for deleting nsx policy from nsxt